gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-extensiongrant/src/main/java/io/gravitee/am/plugins/extensiongrant/core/impl/ExtensionGrantPluginManagerImpl.java

```diff
@@ -18,156 +18,48 @@
 import io.gravitee.am.extensiongrant.api.ExtensionGrant;
 import io.gravitee.am.extensiongrant.api.ExtensionGrantConfiguration;
 import io.gravitee.am.extensiongrant.api.ExtensionGrantProvider;
-import io.gravitee.am.identityprovider.api.AuthenticationProvider;
-import io.gravitee.am.identityprovider.api.NoAuthenticationProvider;
-import io.gravitee.am.plugins.extensiongrant.core.ExtensionGrantConfigurationFactory;
-import io.gravitee.am.plugins.extensiongrant.core.ExtensionGrantDefinition;
 import io.gravitee.am.plugins.extensiongrant.core.ExtensionGrantPluginManager;
-import io.gravitee.plugin.core.api.Plugin;
+import io.gravitee.am.plugins.extensiongrant.core.ExtensionGrantProviderConfiguration;
+import io.gravitee.am.plugins.handlers.api.core.ConfigurationFactory;
 import io.gravitee.plugin.core.api.PluginContextFactory;
-import io.gravitee.plugin.core.internal.AnnotationBasedPluginContextConfigurer;
+import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.InitializingBean;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.ConfigurableApplicationContext;
-import org.springframework.context.annotation.Import;
-
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.*;
 
 /**
  * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
  * @author GraviteeSource Team
  */
-public class ExtensionGrantPluginManagerImpl implements ExtensionGrantPluginManager {
+public class ExtensionGrantPluginManagerImpl extends ExtensionGrantPluginManager {
```

Comment:
I think this should be merged with the abstract class (and made not abstract).

Reply:
The abstraction here is useful for 1 thing:
- Containing the `NamedBeanProcessor` that aren't accessible to the outside
- Not having noise of all the `ProviderPluginManager` generic implementation for reading purpose

I kind of agree right now it is made for "tidying purposes, particurlarly if we want to 
I still believe it's good practice to work with abstractions in case we want to proxy these or have different implementation

Reply:
It's flawed thinking to make abstractions "in case" we need them later, hence the mnemonic YAGNI - you ain't gonna need it. `NamedBeanFactoryPostProcessor` is already contained by inherited `protected` methods. Also, I disagree that having a child of `ProviderPluginManager` itself be abstract solves any readability purposes that a single class would not.